tests/e2etest/GenericTest/src/main/java/org/astarteplatform/devicesdk/tests/e2etest/utilities/GenericDeviceMockDataFactory.java

```diff
@@ -0,0 +1,15 @@
+package org.astarteplatform.devicesdk.tests.e2etest.utilities;
+
+import org.astarteplatform.devicesdk.AstarteDevice;
+
+public class GenericDeviceMockDataFactory implements GenericDeviceMockData {
```

Comment:
Since you only have one implementation of this class you could just avoid the interface and keep this class with static methods that return the fields you need. But since the interface got used pretty heavily throughout the code you can also choose to keep it.

Reply:
I implemented the factory pattern for initializing devices and mock data. I use this pattern to make the code more flexible. In case of changes, it is easier to maintain the original code.